examples/ablation-workshop-mpi.py

```diff
@@ -0,0 +1,763 @@
+            mass=gas_source_term,
+            momentum=cv.momentum*0.0,
+            energy=visc_diss_energy,
+            species_mass=cv.species_mass)  # this should be empty in this case
+
+        return inviscid_rhs + viscous_rhs + source_terms, resin_pyrolysis
+
+    def _rhs(t, state):
+
+        cv, wall_density, tseed = state
+
+        fluid_state = _make_state(cv, gas_model, temperature_seed, wall_density)
+
+        boundaries = make_obj_array([pressure_boundaries, velocity_boundaries])
+
+        cv_rhs, wall_rhs = phenolics_operator(
+            dcoll=dcoll, fluid_state=fluid_state, boundaries=boundaries,
+            gas_model=gas_model, pyrolysis=pyrolysis, quadrature_tag=quadrature_tag,
+            time=t, bprime_class=bprime_class,
+            pressure_scaling_factor=pressure_scaling_factor, penalty_amount=1.0)
+
+        # ~~~~~
+        return make_obj_array([cv_rhs, wall_rhs, tseed*0.0])
+
+    compiled_rhs = actx.compile(_rhs)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    visualizer = make_visualizer(dcoll)
+
+    def my_write_viz(step, t, state):
+
+        cv = state.cv
+        dv = state.dv
+        wdv = gas_model.wall.dependent_vars(dv.wall_density)
+
+        viz_fields = [("CV_density", cv.mass),
+                      ("CV_energy", cv.energy),
+                      ("DV_T", dv.temperature),
+                      ("DV_P", dv.pressure),
+                      ("WV_phase_1", dv.wall_density[0]),
+                      ("WV_phase_2", dv.wall_density[1]),
+                      ("WV_phase_3", dv.wall_density[2]),
+                      ("WDV", wdv)
+                      ]
+
+        # depending on the version, paraview may complain without this
+        viz_fields.append(("x", nodes[0]))
+
+        write_visfile(dcoll, viz_fields, visualizer, vizname=vizname,
+            step=step, t=t, overwrite=True, vis_timer=vis_timer, comm=comm)
+
+    def my_write_restart(step, t, state):
+        cv = state.cv
+        dv = state.dv
+        rst_fname = rst_pattern.format(cname=casename, step=step, rank=rank)
+        if rst_fname != restart_file:
+            rst_data = {
+                "local_mesh": local_mesh,
+                "cv": cv,
+                "wall_density": dv.wall_density,
+                "tseed": dv.temperature,
+                "t": t,
+                "step": step,
+                "order": order,
+                "global_nelements": global_nelements,
+                "nparts": nparts
+            }
+            from mirgecom.restart import write_restart_file
+            write_restart_file(actx, rst_data, rst_fname, comm)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    from warnings import warn
+    warn("Running gc.collect() to work around memory growth issue ")
+    import gc
+    gc.collect()
+
+    my_write_viz(step=istep, t=t, state=fluid_state)
+
+    cv = fluid_state.cv
+    dv = fluid_state.dv
+
+    freeze_gc_flag = True
+    while t < t_final:
```

Comment:
I just added in the most simple possible way. If you think it is necessary, I can add the pre-step, advance state, post-step and so on.

Reply:
Dunno... If you think this example will end up in main eventually in some form, then I guess it would be preferable to make it use `advance_state` like the other examples. But if it's just a temporary thing, it's probably fine to leave it as is.

Reply:
fwiw, I don't mind having some examples that demonstrate "another way".

Reply:
Done in [b9d8385](https://github.com/illinois-ceesd/mirgecom/pull/875/commits/b9d8385d0228402dad0cb5caf387cf5ff3cb0086)